### PR TITLE
gh-64490: Fix refcount error when arguments are packed to tuple in argument clinic

### PR DESCRIPTION
**Pull request by colorfulappl**
While Argument Clinic generated code is packing varargs to a tuple, the arguments' refcounts are not increased.
So all the packed arguments‘ refcounts are decreased 1 improperly when the tuple is released later.

This bug is found in https://github.com/python/cpython/pull/96178 "Argument Clinic functional test".

* Issue: gh-64490

### CODE REVIEW

**Comment by colorfulappl** [comment]
Added Argument Clinic functional test (#96002).
@erlend-aasland @kumaraditya303

**Review by erlend-aasland — APPROVED** [review]
LGTM. Kumar, do you have any remarks?

File: Tools/clinic/clinic.py [hunk 1]
```diff
@@ -956,12 +956,14 @@ def parser_body(prototype, *fields, declarations=''):
                         parser_code.append(normalize_snippet("""
                             %s = PyTuple_New(%s);
                             for (Py_ssize_t i = 0; i < %s; ++i) {{
+                                Py_INCREF(args[%d + i]);
                                 PyTuple_SET_ITEM(%s, i, args[%d + i]);
```
Comment (kumaraditya303) [inline Tools/clinic/clinic.py]:
```suggestion
                                PyTuple_SET_ITEM(%s, i, Py_NewRef(args[%d + i]));
```

**Review by kumaraditya303 — APPROVED** [review]
LGTM

**Comment by kumaraditya303** [comment]
Merged, thanks!

**Comment by colorfulappl** [comment]
Thank you!

**Pull request merged**